File: tests/fixtures/testproject/vbnet/Engine.vb
Imports System.IO

Interface Store
End Interface

Class Base
End Class

Class Settings
End Class

Class Request
End Class

<Serializable>
Public Class Engine
  Inherits Base
  Implements Store
  Private settings As Settings
  
  <Obsolete>
  Public Function Run(req As Request) As Settings
    Dim cfg = File.ReadAllText("config/app.json")
    Return New Settings()
  End Function
End Class

File: tests/fixtures/testproject/kotlin/Engine.kt
package testproject.kotlin

import java.io.File

annotation class Route

open class Base {}
interface Store {}
class Settings {}
class Request {}

@Route
class Engine : Base(), Store {
  val settings: Settings = Settings()
  
  @Route
  fun run(req: Request): Settings {
    val cfg = File("config/app.json").readText()
    return Settings()
  }
}
